### PR TITLE
Add example project

### PR DESCRIPTION
**Pull request by david-gang**
After merging this will add examples of builder:
https://github.com/rharter/auto-value-moshi/pull/139

### CODE REVIEW

File: example/src/main/java/com/ryanharter/auto/value/moshi/example/BirthDateAdapter.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import com.squareup.moshi.FromJson;
+import com.squareup.moshi.ToJson;
+
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Locale;
+
+public class BirthDateAdapter {
```
Comment (ZacSweers) [inline example/src/main/java/com/ryanharter/auto/value/moshi/example/BirthDateAdapter.java]: Can you doc this?

Comment (ZacSweers) [reply example/src/main/java/com/ryanharter/auto/value/moshi/example/BirthDateAdapter.java]: In particular - it seems a bit unnecessary and more of a Moshi example than an auto-value-moshi one

File: example/src/test/java/com/ryanharter/auto/value/moshi/example/PersonTest.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        String json = "{\"name\":\"Piasy\",\"gender\":1,"
+                + "\"age\":23,\"birthdate\":\"" + birthdate
+                + "\",\"address\":{\"street-name\":\"street\",\"city\":\"city\"}}";
+
+        JsonAdapter<Person> jsonAdapter = moshi.adapter(Person.class);
+        String toJson = jsonAdapter.toJson(person);
+        Assert.assertEquals(json, toJson);
+
+        Person fromJson = jsonAdapter.fromJson(json);
+        Assert.assertEquals("Piasy", fromJson.name());
```
Comment (ZacSweers) [inline example/src/test/java/com/ryanharter/auto/value/moshi/example/PersonTest.java]: Static import these methods

**Review by ZacSweers — COMMENTED** [review]
Can you add copyright headers to all the new files?

File: example/src/main/java/com/ryanharter/auto/value/moshi/example/BirthDateAdapter.java (hunk 1 above)
Comment (david-gang) [reply example/src/main/java/com/ryanharter/auto/value/moshi/example/BirthDateAdapter.java]: ok. will remove this

**Comment by david-gang** [comment]
round done

File: example/src/test/java/com/ryanharter/auto/value/moshi/example/PersonTest.java (hunk 2 above)
Comment (david-gang) [reply example/src/test/java/com/ryanharter/auto/value/moshi/example/PersonTest.java]: done

**Comment by david-gang** [comment]
> Can you add copyright headers to all the new files?

I did not find copyright headers on java files. Please look here: https://github.com/rharter/auto-value-moshi/search?q=copyright&unscoped_q=copyright

I don't mind to put a copyrgiht header but what should be the format?

**Comment by david-gang** [comment]
Hi, please let me if something additional is needed from my side 
I think I addressed all comments 
Thanks

**Comment by ZacSweers** [comment]
Sorry we've been busy, we'll circle back soon

**Comment by ZacSweers** [comment]
> I did not find copyright headers on java files.

Sorry! I assumed we did. We'll take care of that in a different PR later. WIll merge this as-is!

**Pull request merged**